time/ptp.md

```diff
@@ -0,0 +1,225 @@
+---
+title: Precision Time Protocol
+authors:
+  - "@zshi-redhat"
+reviewers:
+  - "@oglok"
+  - "@fepan"
+  - "@pliurh"
+approvers:
+  - TBD
+creation-date: 2019-09-03
+last-updated: 2019-09-03
+status: implementable
+---
+
+# Add-Precision-Time-Protocol-Support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+A proposal to enable Precision Time Protocol(PTP) on OpenShift.
+
+The linuxptp package, an implementaion of PTP according to IEEE standard 1588
+for Linux, can be installed on a node and configured to synchronize system
+clock to remote PTP master clock.
+
+## Motivation
+
+NFV cRAN (Centralized Radio Access Networks) and vRAN (Virtual Radio Access
+Networks) require PTP level accuracy in sub-microsecond range, associated
+standards are based on PTP for time and phase synchronization.
+
+### Goals
+
+- Install linuxptp package on baremetal nodes
+- Configure linuxptp services on baremetal nodes
+
+### Non-Goals
+
+- Replace NTP
+- Support PTP software timestamping
+- Support PTP on platforms such as AWS, VMware, OpenStack
```

Comment:
Good one! PTP achieves its high accuracy via hardware time stamping which is a capability in Hardware Interface Card. In virtualization environment, unless Physical Function is passthough to VM, it's impossible for application such as `linuxptp` to talk directly with NIC adapter. The recommended way to sync VM system clock is using KVM virtual PTP driver which allows guest to sync its clock to the host clock with high precision. KVM virtual PTP driver doesn't require any specific hardware support, it can be used in every VM using KVM.

Reply:
The initial proposal is to support PTP with hardware timestamping on baremetal host, not KVM virtual PTP driver. 
Enabling the latter requires:
1) loading kvm_ptp kernel module (can be done via ignition)
2) configuring chrony.conf file (can be done via MCO config)